src/charm.py

```diff
@@ -58,6 +59,7 @@ class IntegratorCharm(CharmBase):
 
     def _setup_database_requirer(self, relation_name: str) -> DatabaseRequires:
         """Handle the creation of relations and listeners."""
+        entity_name, password = self.requested_entities_secret_content
```

Comment:
As is, the relation will proceed even if there's an issue with the secret, but since this is initialised in the `IntegratorCharm`'s constructor, it would be tricky to stop the relation.